internal/graph/check_fast_path.go

```diff
@@ -553,15 +548,14 @@ func constructLeftChannels(
 		}
 		leftChan, err := fastPathRewrite(ctx, r, rel.GetRewrite())
 		if err != nil {
-			// if the resolver already started it needs to be drained
-			if len(leftChans) > 0 {
-				go drainIteratorChannel(fanInIteratorChannels(ctx, leftChans, concurrencyLimit))
-			}
-			return nil, err
+			errCh := make(chan *iterator.Msg, 1)
+			errCh <- &iterator.Msg{Err: err}
+			close(errCh)
```

Comment:
want to verify the logic here.  We have close the channel `errCh` but we still add the channel `err` to `leftChans`?  But since the channel is already closed, TrySendThroughChannel and drainOnExit for the channel will do nothing..

Reply:
It will actually be handled by the consumer. We close it so that the consumer knows there will be no more messages in the channel and can move on to the next one.